### PR TITLE
feat(ui5-button): Implement SAP Horizon theme parameters

### PR DESCRIPTION
**Pull request by NHristov-sap**
CSS parameters have been applied in the component's CSS files for Horizon/Horizon Dark/Horizon HCB/Horizon HCW themes

related to: #5009

### CODE REVIEW

File: packages/main/src/themes/Button.css [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	font-size: var(--sapFontSize);
 	text-shadow: var(--_ui5_button_text_shadow);
 	border-radius: var(--_ui5_button_border_radius);
-	border-width: 0.0625rem;
+	border-width: var(--sapButton_BorderWidth);
 	cursor: pointer;
 	background-color: var(--sapButton_Background);
 	border: 1px solid var(--sapButton_BorderColor);
```
Comment (ilhan007) [inline packages/main/src/themes/Button.css:18]: Should not this:
_border: 1px solid var(--sapButton_BorderColor);_
become:
_border: var(--sapButton_BorderWidth) solid var(--sapButton_BorderColor);_

and perhaps then this becomes obsolete:
_border-width: var(--sapButton_BorderWidth);_

Comment (NHristov-sap) [reply packages/main/src/themes/Button.css]: Done.

File: packages/main/src/themes/sap_belize_hcb/sizes-parameters.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	--_ui5_split_button_middle_separator_top: 0;
+	--_ui5_split_button_middle_separator_height: 2.5rem;
+	--_ui5_split_button_text_button_width: 2.5rem;
+}
+
+[data-ui5-compact-size],
+.ui5-content-density-compact,
+.sapUiSizeCompact {
+	/* SplitButton */
+	--_ui5_split_button_middle_separator_top: 0;
```
Comment (Todor-ads) [inline packages/main/src/themes/sap_belize_hcb/sizes-parameters.css:14]: --_ui5_split_button_middle_separator_top: 0;
Cozy and Compact have same value - no need to add in size parameters

File: packages/main/src/themes/sap_belize_hcw/sizes-parameters.css [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	--_ui5_split_button_middle_separator_top: 0;
+	--_ui5_split_button_middle_separator_height: 2.5rem;
+	--_ui5_split_button_text_button_width: 2.5rem;
+}
+
+[data-ui5-compact-size],
+.ui5-content-density-compact,
+.sapUiSizeCompact {
+	/* SplitButton */
+	--_ui5_split_button_middle_separator_top: 0;
```
Comment (Todor-ads) [inline packages/main/src/themes/sap_belize_hcw/sizes-parameters.css:14]: also here

Comment (Todor-ads) [reply packages/main/src/themes/sap_belize_hcw/sizes-parameters.css:14]: ToggleButton and SegmentedButton (HCB & HCW) - in tog hover & down is not correct color

File: packages/main/src/themes/sap_belize_hcb/sizes-parameters.css (hunk 2 above)
Comment (NHristov-sap) [reply packages/main/src/themes/sap_belize_hcb/sizes-parameters.css:14]: Done

File: packages/main/src/themes/sap_belize_hcw/sizes-parameters.css (hunk 3 above)
Comment (NHristov-sap) [reply packages/main/src/themes/sap_belize_hcw/sizes-parameters.css:14]: Done

Comment (Todor-ads) [reply packages/main/src/themes/sap_belize_hcw/sizes-parameters.css]: Transparent SplitButton in HCB have wrong background color (disabled state)

**Comment by Todor-ads** [comment]
Transparent SplitButton in HCB have wrong background color (disabled state)

**Comment by Todor-ads** [comment]
Emphasized Button and Emphasized SplitButton in HCB/HCW must have 2 px border

**Comment by Todor-ads** [comment]
Emphasized Button (focused) - is not have correct color (white and black) - fiori_3

File: .husky/common.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+command_exists () {
```
Comment (Todor-ads) [inline .husky/common.sh:1]: do you need to change this file?

File: .husky/common.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  command -v "$1" >/dev/null 2>&1
```
Comment (Todor-ads) [inline .husky/common.sh:2]: do you need to change this file?

File: .husky/common.sh (hunk 4 above)
Comment (NHristov-sap) [reply .husky/common.sh:1]: yepp, otherwise there's no way to push

File: .husky/common.sh (hunk 5 above)
Comment (NHristov-sap) [reply .husky/common.sh:2]: same here

**Review by Todor-ads — APPROVED** [review]

File: packages/main/src/themes/Button.css [hunk 6]
```diff
@@ -244,53 +236,56 @@ bdi {
...
-	background-color: var(--sapButton_Active_Background);
+	background-color: var(--sapButton_Lite_Active_Background);
+	border-color: var(--sapButton_Lite_Active_BorderColor);
 	color: var(--sapButton_Active_TextColor);
-	text-shadow: none;
 }
 
-:host([design="Transparent"]:not([active]):not([_is-touch]):hover) {
-	border-color: var(--sapButton_Lite_Hover_BorderColor);
+/* SegmentedButton and ToggleButton */
```
Comment (ilhan007) [inline packages/main/src/themes/Button.css:276]: Is there a reason to add the following styles are added here?
The Button styles are too many to also add styles related to descendants.

Please add them in the respective css files of the components. You can also introduce ui5-togglr-btn-root and ui5-segmented-btn-root (or similar) to distinguish and detach from ui5-button-root.
In this case, it's better to repeat them (if we see that the SegmentedButton and ToggleButton share a lot of common css that we write specifically for them, we can extract these styles to separate file for reuse).

**Pull request merged**